packages/angular-recipe-app/src/app/shopping-list/shopping-list.component.ts

```diff
@@ -18,4 +18,8 @@ export class ShoppingListComponent implements OnInit {
       this.ingredients = ingredients;
     });
   }
+
+  ngOnDestroy(): void {
+    this.shoppingListService.addIngredientEvent.unsubscribe();
```

Comment:
This is very dangerous and we never do such thing. Please read a documentation of EventEmitter and the related unsubsubscribe and maybe you can discuss it with AI what could be the potential problem. When you done with investigation we can discuss your findings.

Reply:
Interesting, have you read my comment? Seems like there is nothing to unsubscribe from then :)

https://github.com/imrepjr/self-development/pull/27#issuecomment-2429235358

Reply:
Also, please give this one below a read if you have the time, I'll schedule a call with you if you'd like to discuss these, I'd be more than happy to.

https://github.com/imrepjr/self-development/issues/19